### PR TITLE
enh: Use gotestsum to provide better unit test output

### PR DESCRIPTION
**Pull request by anhdle-sso**
### Change description

Fixes #

### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: operator/Makefile [hunk 1]
```diff
@@ -37,7 +37,7 @@ all: manager
 # Run tests
 .PHONY: test
 test: generate fmt vet manifests
-	go test -v ./pkg/... -coverprofile cover.out
+	go run gotest.tools/gotestsum@latest --format testname -- -v ./pkg/... -coverprofile cover.out
```
Comment (justinsb) [inline operator/Makefile:40]: TIL that `tools` is a top-level-domain!

**Comment by anhdle-sso** [comment]
@justinsb Please take a look

**Comment by justinsb** [comment]
Thanks @anhdle-sso 

/approve
/lgtm

**Pull request merged**